src/config.ml

```diff
@@ -1297,6 +1318,38 @@ let parse_next (effect:parser_effect) initial_state : (parser_state, 'err) resul
           in
           (* TODO consult `Proto_force *)
           retb (B(Remote, [host, port, proto]))
+        | (`Dev _ | `Dev_type _) as current ->
+          (* there must be a corresponding `Dev or `Dev_type in [tl]*)
+          let typs, tl = List.partition (function
+              | `Dev_type _ -> true
+              | _ -> false) (current::tl) in
+          let names, tl = List.partition (function
+              | `Dev _ -> true
+              | _ -> false) tl in
+          begin match typs, names with
+            | (`Dev_type typ)::_, [`Dev name]
```

Comment:
I'd be more strict, and disallow multiple `dev-type` lines -- i.e. match here on `` | [`Dev_type typ],  [`Dev name]``

Reply:
Originally I always emitted `Dev` and `Dev_type` polyvariants here, but after making `tun0` result in a `Dev` in the gmap directly, this is indeed true - thanks for spotting that! I've revised to do as you suggested.